### DISCUSSION TITLE
Including more tailwind plugins?

### DISCUSSION THREAD

**Discussion by phihung** [dobicinaitis/tailwind-cli-extra#8] [General]
Hi,

Thank you very much for this very useful tool.
Could you consider including more Tailwind plugins in the bundle? For example, the ones listed in [awesome-tailwindcss](https://github.com/aniftyco/awesome-tailwindcss).
We could restrict the update triggers to new Tailwind versions. Possibly in an additional binary release, while keeping the current Tailwind + Daisy setup untouched.

Context: I’m working with the new Python framework, [FastHTML](https://fastht.ml/), and your tool has been a great help. It allows us to avoid installing npm in a Python project altogether.

**Comment by dobicinaitis** [comment]
Hi, I'm really glad that the tool has been useful to you!

The Tailwind CSS team is currently working on the next generation of the framework, which might be released [later this year](https://github.com/tailwindlabs/tailwindcss/discussions/13938#discussioncomment-9938418). According to the [latest status update](https://tailwindcss.com/blog/tailwindcss-v4-alpha), it's still unclear what changes this will bring to the plugin ecosystem and the Standalone CLI:

> * **Plugins and custom utilities** — we don’t have support for plugins, or for writing custom utilities that automatically work with variants yet. Obviously we will make this work before a stable release.
> 
> * **Standalone CLI** — we haven’t worked on a standalone CLI for the new engine yet, but will absolutely have it before the v4.0 release.

Given this uncertainty, I'm hesitant to invest in extending the current solution. If we're fortunate, the new Tailwind CSS Standalone CLI will offer built-in support for third party plugins, potentially making this utility unnecessary. Let's revisit the topic once we'll have more info on the Standalone CLI for Tailwind CSS `v4.0`. 😊

**Comment by phihung** [comment]
Thank you for your detailed answer.
I agree, we should wait for the Tailwind v4.0 release before making further investments.

**Comment by fabge** [comment]
the time has come! 🫡 
https://tailwindcss.com/blog/tailwindcss-v4-beta

**Comment by dobicinaitis** [comment]
Hi! It seems that patching in plugins in Tailwind v4 will be possible and will function similarly to how it works now.

As for adding support for other plugins, frankly it's not something I'm currently interested in pursuing. I originally created this tool for an ex-colleague and haven't yet used it in any of my own projects. As such, spending time extending the functionality of something I don't personally use isn't very appealing.

However, if someone is interested in contributing additional plugin integrations, I'll gladly accept them. For anyone passionate about creating a CLI tool that supports a wide range of Tailwind utilities, I'd recommend forking the project or starting a new one, so they could take all the credit for creating such a tool.

**Comment by fabge** [comment]
hey, i have tackled this and built it from the ground up for tailwindcss v4 and daisyui 5 :)
https://github.com/fabge/tailwindcss-daisyui-cli
